contracts/core/OTCWrapper.sol

```diff
@@ -0,0 +1,691 @@
+import {IOtokenFactoryWrapperInterface} from "../interfaces/otcWrapperInterfaces/IOtokenFactoryWrapperInterface.sol";
+import {MinimalForwarder} from "@openzeppelin/contracts/metatx/MinimalForwarder.sol";
+import {SupportsNonCompliantERC20} from "../libs/SupportsNonCompliantERC20.sol";
+
+/**
+ * @title OTC Wrapper
+ * @author Ribbon Team
+ * @notice Contract that overlays Gamma Protocol for the OTC related interactions
+ */
+contract OTCWrapper is Initializable, OwnableUpgradeable, ReentrancyGuardUpgradeable, ERC2771ContextUpgradeable {
+    using SafeERC20 for IERC20;
+    using SupportsNonCompliantERC20 for IERC20;
+    using SafeMath for uint256;
+
+    AddressBookWrapperInterface public addressbook;
+    MarginRequirementsWrapperInterface public marginRequirements;
+    ControllerWrapperInterface public controller;
+    OracleWrapperInterface public oracle;
+    WhitelistWrapperInterface public whitelist;
+    IOtokenFactoryWrapperInterface public OTokenFactory;
+
+    /************************************************
+     *  EVENTS
+     ***********************************************/
+
+    /// @notice emits an event when an order is placed
+    event OrderPlaced(
+        uint256 indexed orderID,
+        address indexed underlyingAsset,
+        bool isPut,
+        uint256 strikePrice,
+        uint256 expiry,
+        uint256 premium,
+        uint256 notional,
+        address indexed buyer
+    );
+
+    /// @notice emits an event when an order is canceled
+    event OrderCancelled(uint256 orderID);
+
+    /// @notice emits an event when an order is executed
+    event OrderExecuted(
+        uint256 indexed orderID,
+        address collateralAsset,
+        uint256 premium,
+        address indexed seller,
+        uint256 indexed vaultID,
+        address oToken,
+        uint256 initialMargin
+    );
+
+    /// @notice emits an event when collateral is deposited
+    event CollateralDeposited(uint256 indexed orderID, uint256 amount, address indexed acct);
+
+    /// @notice emits an event when collateral is withdrawn
+    event CollateralWithdrawn(uint256 indexed orderID, uint256 amount, address indexed acct);
+
+    /// @notice emits an event when a vault is settled
+    event VaultSettled(uint256 indexed orderID);
+
+    /************************************************
+     *  STORAGE
+     ***********************************************/
+
+    ///@notice order counter
+    uint256 public latestOrder;
+
+    ///@notice fill deadline duration
+    uint256 public fillDeadline;
+
+    ///@notice address that will receive the product fees
+    address public beneficiary;
+
+    /// @notice USDC 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48
+    address public immutable USDC;
+
+    // order status
+    enum OrderStatus {
+        Failed,
+        Pending,
+        Succeeded
+    }
+
+    // struct defining order details
+    struct Order {
+        // underlying asset address
```

Comment:
can you define the decimals of all these values in the comments?

Reply:
https://github.com/ribbon-finance/GammaProtocol-OTC/pull/2/commits/a36609f037c6361686bda62bd70d1550700ffa02